### DISCUSSION TITLE
Ability to assign language aliases in the configuration

### DISCUSSION THREAD

**Discussion by aral** [phcode-dev/phoenix#764] [Q&A] [answered]
It would be very useful (at least for [Kitten](https://codeberg.org/kitten/app) apps) to be able to configure Phoenix to alias certain file types to supported ones.

e.g., 

  - .page → JavaScript
  - .script → JavaScript
  - etc.

Maybe something like:

```json
"aliases": [
  {
    ".page": ".js",
    ".script": ".js"
  }
]
```

You can do this from the drop-down in status bar of the interface and there’s even an option to set it for all files of the given type but that option doesn’t seem to make it sticky or make it apply to other files in the current project even. (At least it didn’t when I was testing it in Ungoogled Chromium.)

**Comment by abose** [comment, marked as answer]
Thanks For the question @aral 
This is already supported via project preferences:
https://github.com/adobe/brackets/wiki/Language-Support#preferences

1. create a file in your project root named `.brackets.json` 
2. Add the following section based on your file extension or name that you have to associate to a language.
```json
{
    "language.fileExtensions": {
        "page": "javascript",
        "script": "javascript"
    },
    "language.fileNames": {
        "someFileName": "python"
    }
}
```

**Comment by aral** [comment]
Thanks. That’s good to know but it would be a hassle to have to do that per project. (I’m thinking of using Phoenix for education so having students configure their editor as step one is the last thing I want to do and hence why I’m looking into a web-based editor.) :) 

Is there any chance this could be considered at an editor preference that’s used on every project?

**Reply by abose** [reply]
It can be added as an editor-wide preference by adding the above json to the editor preferences file
1. Open `debug> open preferences file` and add the above snippet there.

But if you would like to add some well-known extensions to core, I can add that too so that Phoenix recognizes the language extension out of the box.

**Reply by abose** [reply]
Do raise issues/discuss If there is anything that I could help for Education related UX.

We do see heavy use of Phoenix across Educational institutions, but It would be good to have feature requests/ feedback to improve the user experience.

**Comment by r3g3talk** [comment]
Extremely frustrated - the suggestions don't work. I use .shtml files extensively, and cannot get phoenix to show live preview. Just get "Nothing to display" Error page instead. 

PLEASE PLEASE do you have a fix?

Modified both phcode.json, and created a .brackets.json within the project without any luck without any luck.

in the io.phocode folder - file phcode.json:

{
    "fonts.fontSize": "12px",
    "fonts.fontFamily": "'SourceCodePro-Medium', ?? ????, 'MS Gothic', monospace",
    "healthData.healthDataTracking": false,
    "livePreviewShowAtStartup": true,
    "files.associations": {"shtml": "html"},
    "language.fileExtensions": {
        "shtml": "html",
        "shtm": "html"
    },
    "language.fileNames": {
        "shtml": "html"
    }

}

**Reply by abose** [reply]
We have just added shtml support for live preview a few days back https://github.com/phcode-dev/phoenix/pull/1727

It should be available in july release

**Comment by r3g3talk** [comment]
Thanks Arun,

 

I’ve been checking for the newer version with the .shtml fix for windows but have not seen it yet.

**Reply by abose** [reply]
Version 3.9 has this fixed and is rolling out today. check out for the release announcement on twitter or just visit back later today.
